### PR TITLE
[#626] detect if no POM updates are made and exit early

### LINKED ISSUE #626: BUG: Build fails if archetype changes are made without substantive changes to archetype output
## Description
In #624, some improvements were made to the POMs generated by the archetype. The _effective_ POMs (with resolved property values) were the same. This put the build in a state where the archetype test (`mvn clean install -pl :foundation-archetype -Parchetype-test`) was detected as uncached and executed, but the actual resulting test project hitting the build cache and skipping project generation.  Because of the way the test-archetype script works, this resulted in a [build failure](https://github.com/boozallen/aissemble/actions/runs/13979529460/job/39141484759).

## Steps to Reproduce
Clear, specific, and detailed steps taken to enable reproduction of the bug for investigation.
1. Run the archetype test
    ```sh
    mvn clean install -pl :foundation-archetype -Parchetype-test
    ```
2. Run the archetype test again, skipping the build cache to simulate a nominal change
    ```sh
    mvn clean install -pl :foundation-archetype -Parchetype-test -Dmaven.build.cache.skipCache
    ```

### Expected Behavior
The test script detects that the build-cache was hit on the first build after archetype generation (indicating no effective changes) and exits early with success.

### Actual Behavior
The test script injects an invalid module named `*` into a few of the POMs and the subsequent build fails causing the whole test to fail.

## Test Steps
1. Follow steps to reproduce and verify the build succeeds
2. Clear the build-cache for the test project
    ```sh
    rm -rf ~/.m2/build-cache/v1/com.bah.aiops
    ```
3. Unset your docker host to ensure the docker module fails to build
    ```sh
    unset DOCKER_HOST
    ```
4. Run the archetype test to cause the docker failure
    ```sh
    mvn clean install -pl :foundation-archetype -Parchetype-test -Dmaven.build.cache.skipCache
    ```
5. Run the test again and verify it does not succeed due to the build cache
    ```sh
    mvn clean install -pl :foundation-archetype -Parchetype-test -Dmaven.build.cache.skipCache
    ```

## Additional Context
Error output:
```sh
08:39:52 [ERROR] The build could not read 3 projects -> [Help 1]
08:39:52 [ERROR]   
08:39:52 [ERROR]   The project com.bah.aiops:test-generator-pipelines:1.0.0-SNAPSHOT (/Users/ewilkins/dev/aiops/aissemble/foundation/foundation-archetype/target/temp/test-generator/test-generator-pipelines/pom.xml) has 1 error
08:39:52 [ERROR]     Child module /Users/ewilkins/dev/aiops/aissemble/foundation/foundation-archetype/target/temp/test-generator/test-generator-pipelines/* of /Users/ewilkins/dev/aiops/aissemble/foundation/foundation-archetype/target/temp/test-generator/test-generator-pipelines/pom.xml does not exist
08:39:52 [ERROR]   
08:39:52 [ERROR]   The project com.bah.aiops:test-generator-shared:1.0.0-SNAPSHOT (/Users/ewilkins/dev/aiops/aissemble/foundation/foundation-archetype/target/temp/test-generator/test-generator-shared/pom.xml) has 1 error
08:39:52 [ERROR]     Child module /Users/ewilkins/dev/aiops/aissemble/foundation/foundation-archetype/target/temp/test-generator/test-generator-shared/* of /Users/ewilkins/dev/aiops/aissemble/foundation/foundation-archetype/target/temp/test-generator/test-generator-shared/pom.xml does not exist
08:39:52 [ERROR]   
08:39:52 [ERROR]   The project com.bah.aiops:test-generator-docker:1.0.0-SNAPSHOT (/Users/ewilkins/dev/aiops/aissemble/foundation/foundation-archetype/target/temp/test-generator/test-generator-docker/pom.xml) has 1 error
08:39:52 [ERROR]     Child module /Users/ewilkins/dev/aiops/aissemble/foundation/foundation-archetype/target/temp/test-generator/test-generator-docker/* of /Users/ewilkins/dev/aiops/aissemble/foundation/foundation-archetype/target/temp/test-generator/test-generator-docker/pom.xml does not exist
```
Errant behavior captured by script output:
```sh
INFO: Adding 
	<module>*</module>
	to /Users/ewilkins/dev/aiops/aissemble/foundation/foundation-archetype/target/temp/test-generator/test-generator-shared/pom.xml
INFO: Adding 
	<module>*</module>
	to /Users/ewilkins/dev/aiops/aissemble/foundation/foundation-archetype/target/temp/test-generator/test-generator-docker/pom.xml
INFO: Adding 
	<module>*</module>
	to /Users/ewilkins/dev/aiops/aissemble/foundation/foundation-archetype/target/temp/test-generator/test-generator-pipelines/pom.xml
```

### PR DESCRIPTION
**Pull request by epwilkins**
Also removes the test Maven build cache config as it is no longer used as of [PR #482](https://github.com/boozallen/aissemble/pull/506/files#diff-c6e8e0c843d74dbbb6ae6fdedbd8027a96606d60595a4b94117c688f3d97131aL201)

### CODE REVIEW

File: foundation/foundation-archetype/pom.xml [hunk 1]
```diff
@@ -147,7 +147,7 @@
                         <executions>
                             <execution>
                                 <id>test-project-archetype</id>
-                                <phase>generate-resources</phase>
+                                <phase>integration-test</phase>
```
Comment (epwilkins) [inline foundation/foundation-archetype/pom.xml:150]: I: Just felt more correct to have it run in the IT phase. No functional difference.

File: foundation/foundation-archetype/test-project-archetype.sh [hunk 2]
```diff
@@ -250,16 +259,30 @@ echo "{
...
+  updates=true
+fi
 
 cd ../test-generator-pipelines/ || exit
-updatePomBasedOnChildDirs "<!-- TODO: replace with your pipeline-specific modules here -->"
+if updatePomBasedOnChildDirs "<!-- TODO: replace with your pipeline-specific modules here -->"; then
+  updates=true
+fi
+
+if [ $updates = false ]; then
```
Comment (epwilkins) [inline foundation/foundation-archetype/test-project-archetype.sh:282]: I: There is an edge case here where some modules have changes, but others don't. This could easily result in a build failure since there will be missing modules in the subsequent builds. It's difficult for me to conceive of how this scenario comes about, as any change to `foundation-mda` will invalidate the entire project. Because of this, and that it's quite complex to try and account for I'm leaving it intentionally unresolved for now. When it causes an issue we'll have a better idea of how the problem manifests and how to fix it.

**Review by carter-cundiff — APPROVED** [review]
